daos/login.js

```diff
@@ -0,0 +1,66 @@
+const mongoose = require('mongoose');
+const bcrypt = require('bcrypt');
+const User = require('../models/user');
+const Token = require('../models/token');
+const saltRounds = 8;
+
+//Random number generator for creating token
+const {v4:uuidv4} = require('uuid');
+
+//accessing database to create new hashed password and store it with email  
+module.exports.signUp = async (credential) => {
+    // const { email, password } = req.body;
+    let user = await User.findOne({ email : credential.email });
+    if (user) {
+        return false;
+    } 
+        else {
+        credential.password = await bcrypt.hash(credential.password, saltRounds);
+        user = await User.create(credential);
+        return user;
+        }
+  
+}
+
+//accessing database to compare shared password with existing password for a given user id
+module.exports.login = async (credentials) => {
+    const user = await User.findOne({ email : credentials.email }).lean();
+    if (!user) { 
+        return false; 
+    };
+    const passwordMatch = await bcrypt.compare(credentials.password, user.password);
+    if (passwordMatch == false) { 
```

Comment:
```suggestion
    if (!passwordMatch) { 
```
or use `=== false` because if `passwordMatch` were somehow `undefined` or `0` or `''`, this would be true